### ISSUE TITLE
Change to CONTINUE behavior?

### ISSUE THREAD

**Issue by kadrlica** [esheldon/fitsio#117]
I recently updated from `v0.9.8rc2` to `v0.9.11` and I've noticed a difference in how `CONTINUE` cards are being dealt with in headers.

I have a header that looks something like this:
```
OBSTYPE = 'object  '           / Observation type                               
CAMSHUT = 'Open    '           / Camera shutter at exposure start              
PROGRAM = 'Precision Astrometry with DECam: A dynamical study of nearby &'      
CONTINUE  'clusters&'                                                          
CONTINUE  '&' / Current observing orogram                                      
OBSERVER= 'DJN, DJJ'           / Observer name(s)                              
PROPOSER= 'BertinE '           / Proposal Principle Investigator               
PROPID  = '2012B-3001'         / Proposal ID                                    
```

In `v0.9.8rc2` this header read without complaint, though it didn't fill the value of the `PROGRAM` card fully:
```
>>> hdr['PROGRAM'] 
'Precision Astrometry with DECam: A dynamical study of nearby &'
>>> hdr['CONTINUE']
''
```

When I try to read this header with `v0.9.11` I get the following exception
```
/cvmfs/des.opensciencegrid.org/fnal/anaconda2/envs/des17a/lib/python2.7/site-packages/fitsio/fitslib.pyc in set_card(self, card_string)
   4300 
   4301                 if front=='CONTINU':
-> 4302                     raise ValueError("CONTINUE not supported")
   4303 
   4304                 if front=='HISTORY':

ValueError: CONTINUE not supported
```

In contrast, `pyfits` (through astropy) seems to be ok with this header and gives:
```
>>> hdr['PROGRAM']
'Precision Astrometry with DECam: A dynamical study of nearby clusters'
```

I can understand if `fitsio` doesn't deal with `CONTINUE` cards, but is there a way for me to still read this header? My specific suggestion would to print a warning rather than raising an exception.

**Comment by esheldon** [comment]
I do think you are the first person to run into this.

If you have an idea how to make this work, please give it a go.  It probably is easy to get working but I didn't actually have any examples lying around for testing and developing this.

**Comment by kadrlica** [comment]
I don't have a great idea of how it should work. Here are some things that I'm puzzling over:
1) I think that `CONTINUE` is not a FITS standard and that `pyfits` has wrapped it specially. I don't think anything similar has been done for `fitsio`.
2) We don't want to remove `CONTINUE` cards from a header that we read and then write just because `fitsio` doesn't parse them.

I'm worried that just ignoring the `CONTINUE` cards might unintentionally lead to (2), but I guess we were mangling the `CONTINUE` cards before anyway?

I've attached a small example ([continue_example.fits.zip](https://github.com/esheldon/fitsio/files/956522/continue_example.fits.zip)) that has `CONTINUE` statements that are parsed correctly by `pyfits`, incorrectly by `v0.9.8rc2`, and cause an error in `v0.9.11`.

**Comment by esheldon** [comment]
It is only  a convention.  According to this page

https://fits.gsfc.nasa.gov/registry/continue_keyword.html

"FITS readers that do not support this convention should interpret the CONTINUE keywords as being equivalent to a COMMENT keyword. "

**Comment by kadrlica** [comment]
Can we implement something like `_set_as_continue` which would be very similar to [`_set_as_comment`](https://github.com/esheldon/fitsio/blob/9144a55b0119c84f00ac9694565079cf8f9ddbd7/fitsio/fitslib.py#L4355-L4360). Maybe something like:
```
    def _set_as_continue(self):
        value=self._extract_comm_or_hist_value()

        self['class'] = TYP_CONT_KEY
        self['name']  = 'CONTINUE'
        self['value'] =  value
```
Then we'd restructure the `if` statements [here](https://github.com/esheldon/fitsio/blob/9144a55b0119c84f00ac9694565079cf8f9ddbd7/fitsio/fitslib.py#L4301-L4310). If this seems reasonable, then I can try to implement it.

**Comment by esheldon** [comment]
yeah, something like that.

Note what we would do is instead append to the previous added key.

**Comment by kadrlica** [comment]
I thought that for our first attempt we weren't going to implement the `CONTINUE` appending, rather we would just treat these cards as comments (as suggested by the convention for readers that don't implement `CONTINUE`). I would suggest that we split this into two separate issues:

1) Interpret the `CONTINUE` keywords as being equivalent to a `COMMENT` keyword. 
2) Implement support for the `CONTINUE` convention.

**Comment by esheldon** [comment]
OK, please proceed however you see fit.

As part of this effort, please also add an example of this to the header in the unit tests.

**Comment by esheldon** [comment]
I created a branch `continuekey`.  Please use that as your base and submit the pull request against that branch.

**Comment by kadrlica** [comment]
Ok, I'm not ready to break off the full implementation of `CONTINUE` keys, but I can give the comment treatment a try.

**Comment by kadrlica** [comment]
So I started digging into the `cfitsio` libraries and it looks like the pieces could be in place to fully implement the continue behavior. For example, there are functions for reading and writing long strings:
[`fits_read_key_longstr`](https://github.com/kadrlica/fitsio/blob/master/cfitsio3370patch/longnam.h#L138)
[`fits_write_key_longstr`](https://github.com/kadrlica/fitsio/blob/master/cfitsio3370patch/longnam.h#L82)
I guess we would need to pop these into `PyFITSObject_write_string_key` and `PyFITSObject_read_header` (and perhaps elsewhere)?

**Comment by esheldon** [comment]
Is the behaviro exactly the same when writing?  With no performance hit?

For reading, I was reading the cards in `PyFITSObject_read_header`  and then converting appropriately in python.  That function `fits_read_key_longstr` is only if you know it is a string ahead of time.

**Comment by kadrlica** [comment]
I don't know about the relative performance of normal string vs long string writing (at the very least you are writing more cards).

I saw what you were doing in `PyFITSObject_read_header` and couldn't come up with a good solution. The only idea that I had was if we continue to do the conversion in python, then once we know something is a string we call `fits_read_key_longstr` (or a wrapper). Then, we ignore the following `CONTINUE` cards in python. As I said before, the full implementation is beyond my current time investment and I've started #118 to host the full implementation discussion.
